Tests/TuistLoaderTests/Utils/ManifestFileLocatorTests.swift

```diff
@@ -28,22 +39,217 @@ final class ManifestFileLocatorTests: TuistUnitTestCase {
+
+        // Then
+        XCTAssertNotNil(configPath)
+        XCTAssertEqual(paths.last, configPath)
+    }
+
+    func test_locateConfig_where_config_not_exist() throws {
+        // Given
+        _ = try createFiles([
+            "Module01/File01.swift",
+            "Module01/File02.swift",
+            "Module01/File03.swift",
+
+            "Module02/File01.swift",
+            "Module02/File01.swift",
+            "Module02/Subdir01/File01.swift",
+            "Module02/Subdir01/File02.swift",
+
+            "File01.swift",
+            "File02.swift",
+        ])
+
+        // When
+        let configPath = subject.locateConfig(at: try temporaryPath())
+
+        // Then
+        XCTAssertNil(configPath)
+    }
+
+    func test_locateConfig_traversing_where_config_not_exist() throws {
+        // Given
+        let paths = try createFiles([
+            "Module01/File01.swift",
+            "Module01/File02.swift",
+            "Module01/File03.swift",
+
+            "Module02/File01.swift",
+            "Module02/File01.swift",
+            "Module02/Subdir01/File01.swift",
+            "Module02/Subdir01/File02.swift",
+
+            "File01.swift",
+            "File02.swift",
+        ])
+        let locatingPath = paths[5] // "Module02/Subdir01/File01.swift"
+
+        // When
+        let configPath = subject.locateConfig(at: locatingPath)
+
+        // Then
+        XCTAssertNil(configPath)
+    }
+
+    func test_locateSetup() throws {
+        // Given
+        let paths = try createFiles([
+            "Module01/File01.swift",
+            "Module01/File02.swift",
+            "Module01/File03.swift",
+
+            "Module02/File01.swift",
+            "Module02/File01.swift",
+            "Module02/Subdir01/File01.swift",
+            "Module02/Subdir01/File02.swift",
+
+            "File01.swift",
+            "File02.swift",
+            "Setup.swift",
+        ])
+
+        // When
+        let setupPath = subject.locateSetup(at: try temporaryPath())
+
+        // Then
+        XCTAssertNotNil(setupPath)
+        XCTAssertEqual(paths.last, setupPath)
+    }
+
+    func test_locateSetup_traversing() throws {
+        // Given
+        let paths = try createFiles([
+            "Module01/File01.swift",
+            "Module01/File02.swift",
+            "Module01/File03.swift",
+
+            "Module02/File01.swift",
+            "Module02/File01.swift",
+            "Module02/Subdir01/File01.swift",
+            "Module02/Subdir01/File02.swift",
+
+            "File01.swift",
+            "File02.swift",
+            "Setup.swift",
+        ])
+        let locatingPath = paths[5] // "Module02/Subdir01/File01.swift"
+
+        // When
+        let setupPath = subject.locateSetup(at: locatingPath)
+
+        // Then
+        XCTAssertNotNil(setupPath)
+        XCTAssertEqual(paths.last, setupPath)
+    }
+
+    func test_locateSetup_where_setup_not_exist() throws {
+        // Given
+        _ = try createFiles([
```

Comment:
is the underscore necessary? I've had a quick look at the function and I see it's annotated with `@discardableResult`